server/controllers/Investment.js

```diff
@@ -1,31 +1,65 @@
 const { ObjectId } = require('mongoose').Types;
+const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);
 const {User, Project, Investment} = require('../models');
 
-exports.addInvestment = async (req, res) => {
-  const { value, projectId } = req.body;
-  const user = req.user
+
+//I've adjusted this to be called prior to the payment but needs a refactor
+//might be cleaner with promise chains
+const invest = async (userId, projectId, investmentAmount) => {
+  const dollarAmount = toDollarsWithCents(investmentAmount)
 
   try {
     const investment = await Investment.create({
-      user: ObjectId(user._id),
+      user: ObjectId(userId),
       project: ObjectId(projectId),
-      value: parseInt(value)
+      value: dollarAmount
     });
-    await User.updateOne({ _id: user._id }, { $push: { investments: investment._id } });
+    await User.updateOne({ _id: userId }, { $push: { investments: investment._id } });
     await Project.updateOne(
       { _id: projectId },
       {
         $inc: {
           'funding.donorCount': 1,
-          'funding.fundingTotal': parseInt(value)
+          'funding.fundingTotal': dollarAmount
         }
       }
     );
-    res.status(201).send(investment);
+    return investment;
   } catch (err) {
     res.status(400).json({
-      error: "User Investment's could not be updated.",
-      err
+      error: "User Investment's could not be updated."
     });
   }
 };
+
+const toDollarsWithCents = (amount) => parseInt(amount * 100);
+
+exports.makePayment = async (req, res) => {
+  const { userId, projectId, token, investmentAmount } = req.body;
```

Comment:
I think in the other PR I had made a comment about using the userId from req.user provided by the isAuth() middleware instead of body or params. It's more secure. The problem with this is that I can make a payment for someone else under their userId since isAuth only checks if you're logged in. But if you use the id provided by isAuth() you can only ever make payments for the current user id that is logged in

Reply:
Thanks. Great point. This is all done